atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -432,6 +433,17 @@ public InetSocketAddress apply(String endpoint) {
             } catch (Exception e) {
                 numTries++;
                 this.<K>handleException(numTries, hostPool.getHost(), e);
+                if (isRetriableWithBackoffException(e)) {
+                    log.warn("Retrying with backoff a query intended for host {}.", hostPool.getHost(), e);
+                    try {
+                        Thread.sleep(numTries * 1000);
```

Comment:
I'm not emotionally attached to it, I can make it a binary exponential backoff and be one of the cool kids. 

Or did you mean why do the backoff at all? In which case, yes, it's for back-pressure, which should especially help with the case of someone-is-being-way-too-parallel-for-their-small-pool-size and is running up against the pool limit.